domain/globalconfigsync/globalconfig.go

```diff
@@ -17,58 +17,37 @@ package globalconfigsync
 import (
 	"context"
 
-	"github.com/pingcap/tidb/ddl/util"
 	"github.com/pingcap/tidb/util/logutil"
-	"go.etcd.io/etcd/clientv3"
+	pd "github.com/tikv/pd/client"
 	"go.uber.org/zap"
 )
 
-const (
-	globalConfigPath     = "/global/config/"
-	keyOpDefaultRetryCnt = 5
-)
-
-// GlobalConfigSyncer stores global config into etcd.
+// GlobalConfigSyncer is used to sync pd global config.
 type GlobalConfigSyncer struct {
-	etcdCli  *clientv3.Client
-	NotifyCh chan ConfigEntry
+	pd       pd.Client
+	NotifyCh chan pd.GlobalConfigItem
 }
 
-// NewGlobalConfigSyncer returns a new GlobalConfigSyncer.
-func NewGlobalConfigSyncer(etcdCli *clientv3.Client) *GlobalConfigSyncer {
-	return &GlobalConfigSyncer{
-		etcdCli:  etcdCli,
-		NotifyCh: make(chan ConfigEntry, 8),
+// StoreGlobalConfig is used to store global config.
+func (s *GlobalConfigSyncer) StoreGlobalConfig(ctx context.Context, item pd.GlobalConfigItem) error {
+	err := s.pd.StoreGlobalConfig(ctx, []pd.GlobalConfigItem{item})
+	if err != nil {
+		logutil.BgLogger().Error("store global config failed", zap.Error(err))
+		return err
 	}
+	logutil.BgLogger().Info("store global config", zap.Any("Configs", item))
+	return err
 }
 
-// ConfigEntry contain the global config Name and Value.
-type ConfigEntry struct {
-	Name  string
-	Value string
-}
-
-// Notify sends the config entry into notify channel.
-func (c *GlobalConfigSyncer) Notify(name, value string) {
-	c.NotifyCh <- ConfigEntry{Name: name, Value: value}
+// PushGlobalConfigItem pushes global config to internal channel and will be sync into pd's GlobalConfig.
+func (s *GlobalConfigSyncer) PushGlobalConfigItem(globalConfigItem pd.GlobalConfigItem) {
+	s.NotifyCh <- globalConfigItem
 }
 
-// StoreGlobalConfig stores the global config into etcd.
-func (c *GlobalConfigSyncer) StoreGlobalConfig(ctx context.Context, entry ConfigEntry) error {
-	if c.etcdCli == nil {
-		return nil
-	}
-
-	key := globalConfigPath + entry.Name
-	err := util.PutKVToEtcd(ctx, c.etcdCli, keyOpDefaultRetryCnt, key, entry.Value)
-	if err != nil {
-		return err
+// NewGlobalConfigSyncer creates a GlobalConfigSyncer.
+func NewGlobalConfigSyncer(p pd.Client) *GlobalConfigSyncer {
+	return &GlobalConfigSyncer{
+		pd:       p,
+		NotifyCh: make(chan pd.GlobalConfigItem, 100),
```

Comment:
move this function to line #31? normally it should be put after the struct definition.